### PR TITLE
Only monkey patch ActiveRecord::Base when it's been loaded.

### PR DESCRIPTION
**Pull request by parndt**
This prevents the database connection being activated when it is not appropriate.
E.g. asset precompilation with sprockets.

All relevant tests passing for me.
@josevalim Also all of devise tests are passing with this change.

Also, all tests pass with Refinery CMS with this change which is the reason I'm making this change so that asset pre-compilation works out of the box with Heroku.

Thanks! :)

### CODE REVIEW

File: lib/orm_adapter/adapters/active_record.rb [hunk 1]
```diff
@@ -4,89 +4,91 @@
...
-
-    # Do not consider these to be part of the class list
-    def self.except_classes
-      @@except_classes ||= [
-        "CGI::Session::ActiveRecordStore::Session",
-        "ActiveRecord::SessionStore::Session"
-      ]
-    end
+ActiveSupport.on_load(:active_record) do
+  class ActiveRecord::Base
```
Comment (josevalim) [inline lib/orm_adapter/adapters/active_record.rb]: `ActiveSupport.on_load` is instance eval'ed inside AR, this means we don't need to open ActiveRecord::Base here :)

File: lib/orm_adapter/adapters/active_record.rb [hunk 2]
```diff
@@ -4,89 +4,91 @@
...
+    extend OrmAdapter::ToAdapter
+
+    class OrmAdapter < ::OrmAdapter::Base
```
Comment (josevalim) [inline lib/orm_adapter/adapters/active_record.rb]: Can we move this class outside the on_load hook? We could define it outside of the AR like this:

```
module OrmAdapter
  class ActiveRecord < Base
    # ...
  end
end
```

And then inside the load hook we simply do:

```
ActiveSupport.on_load(:active_record) do
  extend ::OrmAdapter::ToAdapter
  self::OrmAdapter = ::OrmAdapter::ActiveRecord
end
```

Wdyt?

Comment (parndt) [inline lib/orm_adapter/adapters/active_record.rb]: I'll take a look right now. I was kinda "ugh" about the original solution too. :)

**Comment by parndt** [comment]
@josevalim I've pushed another commit. Please let me know if you have any problems with it :-)

**Comment by josevalim** [comment]
This looks great. Could you please run Devise tests again? If they look good, I can release a new version right away. Thanks a lot.

**Comment by parndt** [comment]
@josevalim Devise's tests are passing for me using Ruby 1.9.3 on the master & v2.0.4 tag.

**Comment by josevalim** [comment]
Gem released. Should we bump the version requirement in Devise?

**Comment by parndt** [comment]
@josevalim haha, we think alike: plataformatec/devise#1738

If you want to do the same for earlier versions than master please cherry-pick that commit to the appropriate locations :-)

**Pull request merged**